src/query.ts

```diff
@@ -99,7 +142,7 @@ export class Query {
     }
 
     if (!isPlainObject(value)) {
-      (filters[filterName] as Array<unknown>).push(value);
+      (filters[filterName] as Array<unknown>) = (filters[filterName] as Array<unknown>).concat(value);
```

Comment:
To prevent duplicates you can store filter values in a `Set()`. When dumping parameters as an object, I think  you can create a regular array this way: `[...mySet]`.

Reply:
This may worth a test, if there's not one.

Reply:
I don't know if having a Set is really just complicating things since most of our parameters are set through methods and most people know about Arrays but Sets are not just as common place (I don't even know which ES* version sports them) and since our object is kind of _dumb_ it shouldn't keep the user too much from adding superfluous information here.

Reply:
Just some facts:

- `Set` and `Map` have ~98% of engines support AFAIK.
- Our object is not _dumb_, it does validation and should provide a valid, clean query.
- They simplify our code (`has`, `del`, …) and checkings.
- Drawback: we should return arrays and plain objects where we use sets and maps, to be able to serialize as JSON, but it's super easy (`Array.from(myMap)`).

Again: up to you.

Reply:
That's the thing: I'm still not 100% sure if our object is dumb or not. On some points we strip perfectly fitting logic and others we add extra logic. Anyway, if you like the Set(), let's do it.